src/views/design/ScriptDetail/index.tsx

```diff
@@ -234,7 +236,8 @@ const ScriptDetail = withStyles(styles)(
                                     }}
                                     variant="contained"
                                     color="secondary"
-                                    disabled={this.isCreateScriptRoute()}
+                                    disabled={this.isCreateScriptRoute()
+                                        || !checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, 'PUBLIC')}
```

Comment:
Can we already use the security group name of the script in question instead of the fixed value 'PUBLIC'.